src/chunk.c

```diff
@@ -41,12 +47,184 @@ chunk_create_from_tuple(HeapTuple tuple, int16 num_constraints)
 	return chunk;
 }
 
+/*
+ * Align a chunk's hypercube in 'aligned' dimensions.
+ *
+ * After a chunk collision scan, this function is called for each chunk in the
+ * chunk scan context. If a chunk includes slices for an aligned dimension, we
+ * cut the context's hypercube to not overlap with that slice.
+ */
+static bool
+do_dimension_alignment(ChunkScanCtx *scanctx, Chunk *chunk)
+{
+	Hypercube  *cube = scanctx->data;
+	Hyperspace *space = scanctx->space;
+	int			i;
+
+	for (i = 0; i < space->num_dimensions; i++)
+	{
+		Dimension  *dim = &space->dimensions[i];
+		DimensionSlice *chunk_slice,
+				   *cube_slice;
+		int64		coord = scanctx->point->coordinates[i];
+
+		if (!dim->fd.aligned)
+			continue;
+
+		/*
+		 * The chunk might not have a slice for each dimension, so fetch slice
+		 * by dimension ID rather than index
```

Comment:
This function is part of a scan that looks for slices that collide with the hypercube of the chunk we want to create. In other words, we collect only those slices of chunks that actually collide, so some chunks that do not collide in all dimensions will only have a subset of its slices. So, after the scan there are chunks in the scan context that have a full set of slices (these chunks collide with the tentative hypercube) and others that only have a partial set of slices (these chunks do not collide with the tentative hypercube, although they might still be interesting to use because of alignment).

I'll try to clarify the comment.